lib/screens_web/controllers/v2/screen_controller.ex

```diff
@@ -76,6 +76,16 @@ defmodule ScreensWeb.V2.ScreenController do
     |> render("index_multi.html")
   end
 
+  def widget(conn, %{"id" => app_id} = params)
+      when app_id in @app_id_strings do
+    app_id = String.to_existing_atom(app_id)
+
+    conn
+    |> assign(:app_id, app_id)
+    |> assign(:widget_data, if(params["widget"], do: Poison.encode!(params["widget"]), else: nil))
```

Comment:
Separately, should we even send a 200 OK response if the `"widget"` key is missing or false/nil? Consider sending a 400 Bad Request instead since this endpoint requires that key in order to do its job.

That can be done like so: (based off of [this](https://github.com/mbta/screens/blob/master/lib/screens_web/controllers/v2/screen_api_controller.ex#L169-L173))
```ex
conn
|> put_status(:bad_request)
|> text("POST /{screen_id}/widget request must contain a JSON body with `widget` key")
```

Reply:
I checked—setting the `widget_data` assign to `nil` would cause `data-widget-data="<%= @widget_data %>"` in the view template to evaluate to `data-widget-data=""`. On the frontend, our `JSON.parse(getDatasetValue("widgetData"))` would throw an error for that:

```js
> JSON.parse("")
// => Uncaught SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
```

Reply:
Ahhhhh much improved! 🙏